x/merkledb/proof.go

```diff
@@ -138,31 +148,39 @@ func (proof *Proof) Verify(ctx context.Context, expectedRootID ids.ID, tokenSize
 	if len(proof.Path) == 0 {
 		return ErrEmptyProof
 	}
-
+	if proof.Key.hasPartialByte() {
+		return ErrProofKeyPartialByte
+	}
 	if err := verifyProofPath(proof.Path, maybe.Some(proof.Key)); err != nil {
```

Comment:
I know this isn't really related to this PR... But trying to verify the correctness of this function...
```golang
// Returns nil iff all the following hold:
//   - Any node with a partial byte length, should not have a value associated with it
//     since all keys with values are written in complete bytes([]byte).
//   - Each key in [proof] is a strict prefix of the following key.
//   - Each key in [proof] is a strict prefix of [keyBytes], except possibly the last.
//   - If the last element in [proof] is [Key], this is an inclusion proof.
//     Otherwise, this is an exclusion proof and [keyBytes] must not be in [proof].
func verifyProofPath(proof []ProofNode, key maybe.Maybe[Key]) error {
```

The comment is pretty weird imo. I feel like the last bullet isn't something that the function is checking. It's documenting the proof.

Should it be like this instead?
```golang
// Returns nil iff all the following hold:
//   - Any node with a partial byte length, should not have a value associated with it
//     since all keys with values are written in complete bytes([]byte).
//   - Each key in [proof] is a strict prefix of the following key.
//   - Each key in [proof] is a strict prefix of [keyBytes], except possibly the last.
//
// If the last element in [proof] is [Key], this is an inclusion proof.
// Otherwise, this is an exclusion proof and [keyBytes] must not be in [proof].
func verifyProofPath(proof []ProofNode, key maybe.Maybe[Key]) error {
```

Or am I misreading this last bullet?

I'm not really sure that I agree with the sentence as a whole either tbh